### PR TITLE
RVS enumeration fix as per hip

### PR DESCRIPTION
**Pull request by manoj-freyr**
RVS with -g options relied on pci APIs to list devices which is low
level. Fixing it to use hip APIs and to be consistent even with case
of removing some devices using HIP_VISIBLE_DEVICES

after this change:
bash-4.4# ./rvs -g

ROCm Validation Suite (version 0.0.40500)

Supported GPUs available:
0000:1D:00.0 - GPU[ 2 -  6059] Vega 20 (Device 26273)
0000:20:00.0 - GPU[ 3 - 15728] Vega 20 (Device 26273)
0000:24:00.0 - GPU[ 4 - 50220] Vega 20 (Device 26273)
0000:3F:00.0 - GPU[ 5 -  5145] Vega 20 (Device 26273)
0000:43:00.0 - GPU[ 6 - 37877] Vega 20 (Device 26273)
0000:46:00.0 - GPU[ 7 - 47258] Vega 20 (Device 26273)
0000:49:00.0 - GPU[ 8 - 58943] Vega 20 (Device 26273)
bash-4.4# export HIP_VISIBLE_DEVICES=0,1,2,3
bash-4.4# ./rvs -g

ROCm Validation Suite (version 0.0.40500)

Supported GPUs available:
0000:1D:00.0 - GPU[ 2 -  6059] Vega 20 (Device 26273)
0000:20:00.0 - GPU[ 3 - 15728] Vega 20 (Device 26273)
0000:24:00.0 - GPU[ 4 - 50220] Vega 20 (Device 26273)
0000:3F:00.0 - GPU[ 5 -  5145] Vega 20 (Device 26273)
bash-4.4# export HIP_VISIBLE_DEVICES=0,1,2,3,4,5
bash-4.4# ./rvs -g

ROCm Validation Suite (version 0.0.40500)

Supported GPUs available:
0000:1D:00.0 - GPU[ 2 -  6059] Vega 20 (Device 26273)
0000:20:00.0 - GPU[ 3 - 15728] Vega 20 (Device 26273)
0000:24:00.0 - GPU[ 4 - 50220] Vega 20 (Device 26273)
0000:3F:00.0 - GPU[ 5 -  5145] Vega 20 (Device 26273)
0000:43:00.0 - GPU[ 6 - 37877] Vega 20 (Device 26273)
0000:46:00.0 - GPU[ 7 - 47258] Vega 20 (Device 26273)

unset HIP_VISIBLE_DEVICES
bash-4.4# ./rvs -g

ROCm Validation Suite (version 0.0.40500)

Supported GPUs available:
0000:1D:00.0 - GPU[ 2 -  6059] Vega 20 (Device 26273)
0000:20:00.0 - GPU[ 3 - 15728] Vega 20 (Device 26273)
0000:24:00.0 - GPU[ 4 - 50220] Vega 20 (Device 26273)
0000:3F:00.0 - GPU[ 5 -  5145] Vega 20 (Device 26273)
0000:43:00.0 - GPU[ 6 - 37877] Vega 20 (Device 26273)
0000:46:00.0 - GPU[ 7 - 47258] Vega 20 (Device 26273)
0000:49:00.0 - GPU[ 8 - 58943] Vega 20 (Device 26273)

### CODE REVIEW

File: pesm.so/src/action.cpp [hunk 1]
```diff
@@ -229,7 +229,7 @@ int pesm_action::run(void) {
  * @return 0 - success. non-zero otherwise
  *
  * */
-int pesm_action::do_gpu_list() {
+int ddddo_gpu_list() {
```
Comment (jkottiku) [inline pesm.so/src/action.cpp:232]: Do we still need this function as its replaced by display_gpu_info() ?

File: src/rvs_util.cpp [hunk 2]
```diff
@@ -180,3 +181,68 @@ bool fetch_gpu_list(int hip_num_gpu_devices, map<int, uint16_t>& gpus_device_ind
     return amd_gpus_found;
 }
 
+
+int display_gpu_info(){
```
Comment (jkottiku) [inline src/rvs_util.cpp:185]: Can we move this function to gpu utils file ?

File: include/rvs_util.h [hunk 3]
```diff
@@ -124,4 +124,5 @@ void *json_node_create(std::string module_name, std::string action_name,
 bool fetch_gpu_list(int hip_num_gpu_devices, map<int, uint16_t>& gpus_device_index,
                 const std::vector<uint16_t>& property_device,
                 const int& property_device_id, bool property_device_all, bool mcm_check = false);
+int display_gpu_info();
```
Comment (jkottiku) [inline include/rvs_util.h:127]: Please add void.
int display_gpu_info(void);

Comment (manoj-freyr) [reply include/rvs_util.h]: done

File: pesm.so/src/action.cpp (hunk 1 above)
Comment (manoj-freyr) [reply pesm.so/src/action.cpp]: done

File: src/rvs_util.cpp (hunk 2 above)
Comment (manoj-freyr) [reply src/rvs_util.cpp]: I thought of that but then it is not exactly a gpu util but something that uses gpu utilities(like dom2node etc) to do something specific to how RVS wants to display the gpus. Hence kept it in rvs_util

Comment (jkottiku) [reply src/rvs_util.cpp]: Ok

**Review by jkottiku — APPROVED** [review]

**Comment by manoj-freyr** [comment]
merging changes.

**Pull request merged**